### PR TITLE
refactor: marketplace custom errors handling

### PR DESCRIPTION
**Pull request by AuHau**
Implementation for https://github.com/codex-storage/codex-contracts-eth/pull/141

### CODE REVIEW

File: tests/contracts/testContracts.nim [hunk 1]
```diff
@@ -118,7 +117,13 @@ ethersuite "Marketplace contracts":
...
+    try:
+      discard await marketplace
+        .markProofAsMissing(slotId, missingPeriod)
+        .confirm(1)
+
+      raise newException(CatchableError, "Should have failed")
+    except Marketplace_SlotNotAcceptingProofs:
+      discard
+    except CatchableError:
+      assert false
```
Comment (markspanbroek) [inline tests/contracts/testContracts.nim:129]:
```suggestion
    expect Marketplace_SlotNotAcceptingProofs:
      discard await marketplace.markProofAsMissing(slotId, missingPeriod).confirm(1)
```

**Review by markspanbroek — APPROVED** [review]
Looks good @AuHau, thanks!

File: tests/contracts/testContracts.nim [hunk 2]
```diff
@@ -118,7 +117,13 @@ ethersuite "Marketplace contracts":
     switchAccount(client)
     let missingPeriod = periodicity.periodOf(await ethProvider.currentTime())
     await ethProvider.advanceTime(periodicity.seconds)
-    check await marketplace
-      .markProofAsMissing(slotId, missingPeriod)
-      .confirm(1)
-      .reverts("Slot not accepting proofs")
+    try:
```
Comment (emizzle) [inline tests/contracts/testContracts.nim:120]: This test doesn't really cover the changes introduced, although I don't know how much we really want to cover *all* of them. We could add a test that calls `requestState` with a non-existent `requestId`, which should return `none RequestState`.
